logging/jboss/src/test/java/io/quarkus/ts/logging/jboss/DefaultMinLogLevelIT.java

```diff
@@ -25,4 +44,35 @@ public void checkDefaultLogMinLevel() {
         // the value of minimum logging level overrides the logging level
         app.logs().assertDoesNotContain("Trace log example");
     }
+
+    @Test
+    @Tag("https://github.com/quarkusio/quarkus/issues/40016")
+    public void checkLogRotationContent() throws IOException {
+        Path logDir = app.getServiceFolder();
+        // one main log file plus others rotated
+        int logFilesNumber = Integer.parseInt(app.getProperty("quarkus.log.file.rotation.max-backup-index").get()) + 1;
+
+        app.given().when().get("/log").then().statusCode(204);
+
+        Awaitility.await().atMost(10, TimeUnit.SECONDS).until(() -> {
+            File[] rotatedFiles = logDir.toFile().listFiles((dir, name) -> name.startsWith(LOG_FILE_NAME));
+            return rotatedFiles != null && rotatedFiles.length == logFilesNumber;
+        });
+
+        // Verify that the rotated file with the expected suffix exists
+        String expectedSuffix = "." + DATE_FORMATTER.format(LocalDate.now()) + "\\.\\d+$";
+        List<Path> rotatedFiles = Files.list(logDir)
+                .filter(path -> path.getFileName().toString().matches(LOG_FILE_NAME + expectedSuffix)).toList();
+
+        assertFalse(rotatedFiles.isEmpty(), "Rotated log files with expected suffix were not found.");
```

Comment:
For record @jedla97 , I still think this condition doesn't make sense. First there is waiting for `max-backup-index` + 1=3 and now it's satisfied with 1